src/vivarium_gates_mncnh/components/tree.py

```diff
@@ -49,3 +61,55 @@ def transition_side_effect(self, index: pd.Index, _event_time: ClockTime) -> Non
         pop = self.population_view.get(index)
         pop[self.update_column] = self.update_value
         self.population_view.update(pop)
+
+
+class ANCInitialState(State):
+    def __init__(
+        self,
+        state_id: str,
+        allow_self_transition: bool = False,
+        initialization_weights: DataInput = 0.0,
+    ) -> None:
+        super().__init__(state_id)
+        self.transition_set = ANCTransitionSet(
+            self.state_id, allow_self_transition=allow_self_transition
+        )
+        self._sub_components = [self.transition_set]
+
+
+class ANCTransitionSet(TransitionSet):
+    def setup(self, builder: Builder) -> None:
+        super().setup(builder)
+        self.propensity = builder.value.get_value(f"antenatal_care.correlated_propensity")
+        self.transitions = self.get_ordered_transitions(builder, self.transitions)
+
+    def choose_new_state(
+        self, index: pd.Index[int]
+    ) -> tuple[list[State | str], pd.Series[Any]]:
+        """Use propensities to choose new state."""
+        outputs, probabilities = zip(
+            *[
+                (transition.output_state, np.array(transition.probability(index)))
+                for transition in self.transitions
+            ]
+        )
+        probabilities = np.transpose(probabilities)
+        outputs, probabilities = self._normalize_probabilities(outputs, probabilities)
+        cdf_probabilities = [np.cumsum(probs) for probs in probabilities]
+        propensities = self.propensity(index)
+        # find index of first cumulative probability that is greater than propensity
+        output_indexes = [
```

Comment:
Can't you leverage [`_choice()`](https://github.com/ihmeuw/vivarium/blob/main/src/vivarium/framework/randomness/stream.py#L382) to do this rather than writing new logic?